packages/manager/apps/telecom/src/app/telecom/pack/migration/offers/pack-migration-offers.controller.js

```diff
@@ -179,13 +184,42 @@ export default class TelecomPackMigrationOffersCtrl {
       });
   }
 
-  getModemOptionLabel(modemOption) {
-    return this.$translate.instant(
-      `telecom_pack_migration_modem_${modemOption.name}`,
-      {
-        price: modemOption.price ? modemOption.price.text : '',
-      },
-    );
+  initializeModemOptions() {
+    this.process.migrationOffers.result.offers.forEach((offer) => {
+      Object.values(offer.modemOptions).map((option) => {
+        let optionLabel = '';
+        if (!offer.modemRental) {
+          optionLabel = this.$translate.instant(
+            `telecom_pack_migration_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        } else if (
+          this.MODEM_LIST.includes(option.name) &&
+          offer.modemRental.value === option.price.value
+        ) {
+          optionLabel = this.$translate.instant(
+            'telecom_pack_migration_modem_keep',
+            { price: option.price ? option.price.text : '' },
+          );
+        } else {
+          optionLabel = this.$translate.instant(
+            `telecom_pack_migration_modem_${option.name}`,
+            {
+              price: option.price ? option.price.text : '',
+            },
+          );
+        }
+        set(option, 'label', optionLabel);
+        return option;
+      });
```

Comment:
**nitpick:** `Array.prototype.map` returns a new array. But its return value is not used/captured. Please use appropriate array method.